File: backend/app/schemas/session.py
from typing import Optional, Dict, Any
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict


SESSION_SOURCE_VALUES = "^(manual|scheduled|market_watch|stop_loss|take_profit)$"


# 基本会话模型
class SessionBase(BaseModel):
    user_id: Optional[int] = None
    stock_code: str
    trading_frequency: str
    trading_strategy: str
    source: str = Field(default="manual", pattern=SESSION_SOURCE_VALUES)


# 创建会话模型
class SessionCreate(SessionBase):
    pass


# 更新会话模型
class SessionUpdate(BaseModel):
    status: Optional[str] = Field(None, pattern="^(active|completed|archived)$")
    stock_code: Optional[str] = None
    source: Optional[str] = Field(None, pattern=SESSION_SOURCE_VALUES)


# 会话响应模型
class SessionResponse(SessionBase):
    session_id: UUID
    created_at: datetime
    updated_at: datetime
    ended_at: Optional[datetime] = None
    status: str
    stock_name: Optional[str] = None

    # 关联数据ID
    data_snapshot_id: Optional[UUID] = None
    debate_thread_id: Optional[UUID] = None

    # 会话上下文
    current_position: Optional[Dict[str, Any]] = None

    model_config = ConfigDict(from_attributes=True)


class SessionListResponse(BaseModel):
    """会话分页列表响应。"""

    total: int
    items: list[SessionResponse]
    limit: int
    skip: int

File: frontend/src/api/session.ts
import { apiClient } from './client';

export interface Session {
  session_id: string;  // 后端使用session_id,不是id
  id?: string;  // 保持向后兼容
  stock_code: string;
  stock_name: string;
  status: 'active' | 'completed' | 'failed';
  created_at: string;
  updated_at: string;
  ended_at?: string | null;
  trading_frequency: string;
  trading_strategy: string;
  source: 'manual' | 'scheduled' | 'market_watch' | 'stop_loss' | 'take_profit';
}

export interface SessionListResponse {
  total: number;
  items: Session[];
  limit: number;
  skip: number;
}

export interface SessionListParams {
  status?: string;
  source?: Session['source'];
  q?: string;
  skip?: number;
  limit?: number;
}

export const sessionApi = {
  create: (data: {
    stock_code: string;
    stock_name?: string;
    trading_frequency: string;
    trading_strategy: string;
    source?: Session['source'];
  }) =>
    apiClient.post<Session>('/sessions/', data),

  list: (params?: SessionListParams) =>
    apiClient.get<Session[]>('/sessions/', { params }),

  listPaginated: async (params?: SessionListParams): Promise<SessionListResponse> => {
    const response = await apiClient.get<SessionListResponse | Session[]>('/sessions/', {
      params: { ...params, paginated: true },
    });
    if (Array.isArray(response)) {
      return {
        total: response.length,
        items: response,
        limit: params?.limit ?? response.length,
        skip: params?.skip ?? 0,
      };
    }
    return response;
  },

  get: (id: string) =>
    apiClient.get<Session>(`/sessions/${id}`),

  update: (id: string, data: Partial<Session>) =>
    apiClient.put<Session>(`/sessions/${id}`, data),

  delete: (id: string) =>
    apiClient.delete<{ message?: string }>(`/sessions/${id}`),

  batchDelete: (sessionIds: string[]) =>
    apiClient.post<{ message?: string }>('/sessions/batch-delete', { session_ids: sessionIds }),
};
